Utiliser uniquement la gem "debug" native

francois-ferrandis opened the pull request:
Nous avons eu une [discussion entre devs sur les outils de debug](https://mattermost.incubateur.net/betagouv/pl/ikmgr3ih9pgduysx8xho891r9r) dans laquelle personne n'exprimait de préférence pour un outil de debug interactif spécifique. 

Je propose donc d'unifier nos pratiques autour de la gem `debug`. Cette gem semble la  plus moderne, beaucoup plus performante que `byebug`, et [suggérée par Rails](https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem). 

En homogénéisant nos pratiques au sein de l'équipe, on a une opportunité de construire une culture technique commune.

La doc de la gem : https://github.com/ruby/debug

Note : une feature me plait particulièrement : 

```
 - catch <Error>
   - Set breakpoint on raising <Error>.
```

# Dans cette PR

J'ai cherché et remplacé les exemples d'usages de `byebug` et `pry` dans notre codebase.

J'ai mis à jour la super doc d'Adrien sur comment on fait du debug pas-à-pas dans les specs JS, j'ai bien vérifié que la gem `debug` permettait d'appeler `click_button "Continuer"` en direct live. :ok_hand:

francois-ferrandis committed:
Utiliser uniquement la gem "debug" native

francois-ferrandis commented on scripts/webhook_debug_rack_app.ru:
```diff
@@ -13,7 +13,7 @@ class WebhookDebugRackApp
     puts "computed_signature : #{computed_signature}"
     puts "received_signature : #{received_signature}"
-    [200, { "Content-Type" => "text/html" }, ["Hello World!"]]
+    [200, { "content-type" => "text/html" }, ["Hello World!"]]
```
Pour éviter une erreur causée par https://github.com/rack/rack/issues/1592.

@adipasquale Tu es l'auteur de ce script, je ne l'ai jamais utilisé, je te laisse le supprimer si tu as envie. 😉

adipasquale reviewed (approved):
❤️ merci d’avoir clarifié et exploré tout ça

hâte de tester

victormours reviewed (approved):
ça me va très bien 👍

AntoineGirard reviewed (approved):
LGTM 🚀

francois-ferrandis committed:
Quelle spec est en timeout?

francois-ferrandis committed:
Et à 30 secondes ?

francois-ferrandis committed:
C'est l'autodoc ?

francois-ferrandis committed:
C'est vraiment l'autodoc ?

francois-ferrandis committed:
byebug CI 1

francois-ferrandis committed:
byebug CI 2

francois-ferrandis committed:
pry CI 1

francois-ferrandis committed:
pry CI 2

francois-ferrandis committed:
debug CI 1

francois-ferrandis committed:
debug CI 2

francois-ferrandis committed:
Assez d'essais (le groupe de rock)

francois-ferrandis committed:
Update parallel_tests to try to prevent hanging forever

francois-ferrandis committed:
Merge remote-tracking branch 'origin/production' into frf/require-debug-gem

francois-ferrandis committed:
OK j'ai trouvé la cause :

https://github.com/ruby/debug/issues/797

Même Rails recommande ça !
https://github.com/rails/rails/pull/51692

francois-ferrandis commented on Gemfile:
```diff
@@ -215,6 +213,10 @@ group :development do
 end
 
+group :development, :test do
+  gem "debug", require: "debug/prelude"
```
Voir https://github.com/ruby/debug/issues/797

C'est aussi la recommandation de Rails : 
https://github.com/rails/rails/pull/51692
